### PR TITLE
#170766106 update trip request functionality

### PR DESCRIPTION
**Pull request by KKDoddy**
### What does this PR do
It adds a functionality of updating a trip request.

- create a middle-ware that checks if the id in the route is valid and that there is a trip request for that id.
- create a middle-ware that checks if the update request is coming from the owner of the trip request.
-  create a middle-ware that checks if the trip request is still open and can be edited.
- include the check remember me functionality to the route.
- create a controller function that updates a trip request

### How should this be manually tested
clone the repository then run server and check the swagger documentation
### What are the relevant pivotal tracker stories
[#170766106](https://www.pivotaltracker.com/story/show/170766106)

### CODE REVIEW

**Review by karengir — APPROVED** [review]

**Review by jajajaden01 — APPROVED** [review]

File: server/controllers/tripController.js [hunk 1]
```diff
@@ -50,6 +62,43 @@ class TripController {
...
+      date,
+      reasons,
+      accommodationId,
+      userId,
+      status
+    };
+
+    tripUpdate = await TripController.setTripType(tripUpdate, returnDate, cityNumber);
+    const updatedTrip = await tripHelpers.updateTrip(tripUpdate, tripId);
+    console.log(tripId);
```
Comment (luc-tuyishime) [inline server/controllers/tripController.js]: Remove console.log

Comment (KKDoddy) [reply server/controllers/tripController.js]: I have removed it and others

**Review by izabayogit — APPROVED** [review]

**Review by luc-tuyishime — APPROVED** [review]

File: server/controllers/tripController.js [hunk 2]
```diff
@@ -50,6 +61,41 @@ class TripController {
       data: saveTrip
     });
   }
+
+  /**
+   * This method handle oneWayTrip request.
```
Comment (istafrica-dennw) [inline server/controllers/tripController.js:66]: Hi @KKDoddy , "This method handle oneWayTrip request.", but I see this "tripType, from, to, date, reasons, accommodationId, returnDate"

File: server/controllers/tripController.js [hunk 3]
```diff
@@ -50,6 +61,41 @@ class TripController {
...
+    } = req.body;
+    const cityNumber = to.length;
+    let tripUpdate = {
+      tripType,
+      from,
+      to,
+      date,
+      reasons,
+      accommodationId,
+      userId,
```
Comment (istafrica-dennw) [inline server/controllers/tripController.js:86]: Are you editing The userId of the Trip?

File: server/controllers/tripController.js [hunk 4]
```diff
@@ -50,6 +61,41 @@ class TripController {
...
+  static async editTrip(req, res) {
+    const tripId = req.oldTrip.id;
+    const userId = req.user.id;
+    const status = 'Pending';
+    const {
+      tripType, from, to, date, reasons, accommodationId, returnDate
+    } = req.body;
+    const cityNumber = to.length;
+    let tripUpdate = {
+      tripType,
```
Comment (istafrica-dennw) [inline server/controllers/tripController.js:80]: Are you changing the TripType? it should be detected from your Data not from the user Entries

**Pull request merged**